rest/main.go

```diff
@@ -305,6 +307,14 @@ func sanitizeDbConfigs(configMap DbConfigMap) (DbConfigMap, error) {
 
 		// Make sure any updates are written back to the config
 		configMap[dbName] = dbConfig
+
+		if _, ok := processedBucketNames[*dbConfig.Bucket]; ok {
+			return nil, fmt.Errorf("automatic upgrade to persistent config failed. Only one database can " +
+				"target any given bucket")
+		}
+
+		processedBucketNames[*dbConfig.Bucket] = struct{}{}
```

Comment:
I'm guessing we don't want to modify the given `configMap` before we've done the bucket check, even if the caller is going to be using the error to stop processing.

```suggestion
		if _, ok := processedBucketNames[*dbConfig.Bucket]; ok {
			return nil, fmt.Errorf("automatic upgrade to persistent config failed. Only one database can " +
				"target any given bucket")
		}
		processedBucketNames[*dbConfig.Bucket] = struct{}{}

		// Make sure any updates are written back to the config
		configMap[dbName] = dbConfig
```

Reply:
Figured it didn't really make a difference but sure.

Reply:
The only reason it does is that this function is modifying the input map, rather than returning a copy.

We can't guarantee the caller is always going to be checking the error.